### PR TITLE
Upload container runtime log to sd/es.

### PR DESCRIPTION
**Pull request by Random-Liu**
I've verified this in my environment. My stackdriver has an extra `container-runtime` entry for node log, and it collects container runtime daemon log correctly.

@yujuhong @feiskyer @crassirostris @piosz 
@kubernetes/sig-node-pr-reviews @kubernetes/sig-instrumentation-pr-reviews 
Signed-off-by: Lantao Liu <lantaol@google.com>

**Release note**:

```release-note
Container runtime daemon (e.g. dockerd) logs in GCE cluster will be uploaded to stackdriver and elasticsearch with tag `container-runtime`
```

### CODE REVIEW

**Comment by Random-Liu** [comment]
@crassirostris Please note that, with this there will be both `docker` tag and `container-runtime` tag in GKE. And for docker as container runtime, their content will be the same. Is this acceptable?

**Comment by Random-Liu** [comment]
/test pull-kubernetes-unit
--

**Comment by coffeepac** [comment]
@Random-Liu this should rev the version of the fluentd-es addon container image.  need to update:
- cluster/addons/fluentd-elasticsearch/fluentd-es-image/Makefile#L19 
- cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml#L51  (and four other locations in this file)

current value is v2.0.4, I believe that v2.0.5 is appropriate.

**Comment by igorpeshansky** [comment]
`fluentd-gcp` actually expects the tag to be in a certain format and parses it to extract container identification. So you can't just use an arbitrary tag -- the logs will not be attributed to the correct container.

**Comment by yujuhong** [comment]
> fluentd-gcp actually expects the tag to be in a certain format and parses it to extract container identification. So you can't just use an arbitrary tag -- the logs will not be attributed to the correct container.

@igorpeshansky the tag is added for the container runtime (e.g., docker daemon) log itself, which does not map to any "container". Does the same rule still apply?

**Comment by igorpeshansky** [comment]
Well, the question is which [resource](https://cloud.google.com/logging/docs/api/v2/resource-list) in Stackdriver the log is going to be associated with. By default, the logging agent expects the tag to be in a [certain format](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/lib/fluent/plugin/out_google_cloud.rb#L267) to extract the resource information -- if the tag doesn't match, it won't be able to infer the resource, and there is currently no easy way to supply one directly to the agent.

**Comment by Random-Liu** [comment]
@igorpeshansky This tag `container-runtime` is for container runtime daemon log, e.g. docker log, containerd log. It is not for container log. The `certain format` you linked seems to be tag for container log?

**Comment by yujuhong** [comment]
> @igorpeshansky This tag container-runtime is for container runtime daemon log, e.g. docker log. It is not for container log. The certain format you linked seems to be tag for container log?

@igorpeshansky I also didn't find anything special usage of the "docker" tag in the files you linked. If that's the case, does fluentd simply pass the tag to stackdriver?

**Comment by igorpeshansky** [comment]
Right, in that case fluentd will default to the instance resource and pass the tag to Stackdriver. There will be no association with the Kubernetes node or any other Kubernetes entity.

**Comment by Random-Liu** [comment]
> Right, in that case fluentd will default to the instance resource and pass the tag to Stackdriver. There will be no association with the Kubernetes node or any other Kubernetes entity.

Can I understand this as for docker daemon log, either `docker` tag or `container-runtime` tag, it doesn't make difference to stackdriver?

Do we assume that there is a `docker` log tag in stackdriver? If not, it seems fine to gradually switch to `container-runtime` tag. :)

**Comment by yujuhong** [comment]
> Right, in that case fluentd will default to the instance resource and pass the tag to Stackdriver. There will be no association with the Kubernetes node or any other Kubernetes entity.

I see. The logs are only associated with the GCE VM instances, and these logs will have two tags (`container-runtime` and `docker`) for the time being.

**Comment by igorpeshansky** [comment]
There is no `docker` tag in Stackdriver — unknown tags are translated into log names, which are per-project and user-defined. Feel free to use either.

**Comment by Random-Liu** [comment]
@igorpeshansky Thanks a lot for the information. :)

**Comment by Random-Liu** [comment]
@coffeepac This is only a config change. Do we really want to build a new image with new version?

**Comment by Random-Liu** [comment]
@coffeepac Updated from 0.1.3 to 0.1.4

**Comment by Random-Liu** [comment]
Apply LGTM based on https://github.com/kubernetes/kubernetes/pull/59103#issuecomment-363951096

**Comment by coffeepac** [comment]
@Random-Liu looks good!  thanks.

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -2041,6 +2041,11 @@ function start-fluentd-resource-update {
   wait-for-apiserver-and-update-fluentd ${fluentd_gcp_yaml} &
 }
 
+# Update {{ container-runtime }} with actual container runtime name.
+function update-container-runtime {
+  sed -i -e "s@{{ *container_runtime *}}@${CONTAINER_RUNTIME_NAME:-docker}@g" "$1"
```
Comment (crassirostris) [inline cluster/gce/gci/configure-helper.sh]: Please make a local variable from the argument, so it's much easier to read this code later (i.e. `local -r configmap_path="$1"`)

**Review by crassirostris — CHANGES_REQUESTED** [review]
Thanks a lot for the change!

Please bump fluentd-gcp configmap version & both for fluentd-gcp and fluentd-es sync configmap version in the configmap manifest file & in the damonset manifest file

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (Random-Liu) [reply cluster/gce/gci/configure-helper.sh]: Done.

**Comment by Random-Liu** [comment]
> Please bump fluentd-gcp configmap version & both for fluentd-gcp and fluentd-es sync configmap version in the configmap manifest file & in the damonset manifest file

Done. fluentd-es-config is changed to `0.1.4`. fluentd-gcp-config is changed to `1.2.4`.

File: cluster/gce/config-default.sh [hunk 2]
```diff
@@ -82,7 +82,8 @@ NODE_IMAGE_PROJECT=${KUBE_GCE_NODE_PROJECT:-cos-cloud}
 NODE_SERVICE_ACCOUNT=${KUBE_GCE_NODE_SERVICE_ACCOUNT:-default}
 CONTAINER_RUNTIME=${KUBE_CONTAINER_RUNTIME:-docker}
 CONTAINER_RUNTIME_ENDPOINT=${KUBE_CONTAINER_RUNTIME_ENDPOINT:-}
-LOAD_IMAGE_COMMAND=${KUBE_LOAD_IMAGE_COMMAND:-docker load -i}
+CONTAINER_RUNTIME_NAME=${KUBE_CONTAINER_RUNTIME_NAME:-}
+LOAD_IMAGE_COMMAND=${KUBE_LOAD_IMAGE_COMMAND:-}
```
Comment (crassirostris) [inline cluster/gce/config-default.sh]: Why is this parameters changed? I don't understand how it's related to the rest of the changes

**Review by crassirostris — COMMENTED** [review]
One nit, otherwise LGTM, thanks for the change!

File: cluster/gce/config-default.sh (hunk 2 above)
Comment (crassirostris) [reply cluster/gce/config-default.sh]: OK, thanks for the explanation!

**Review by crassirostris — APPROVED** [review]
/lgtm

**Pull request merged**